dali/kernels/test/resampling_test/separable_3D_test.cu

```diff
@@ -0,0 +1,542 @@
+    in_shapes[1] = {{
+      { 10, 200, 120, 3 },
+      { 100, 10, 10, 3 },
+      { 70, 80, 90, 3 },
+    }};
+
+    out_shapes[1] = {{
+      { 31, 200, 120, 3 },
+      { 51, 27, 33, 3 },
+      { 73, 181, 43, 3 },
+    }};
+
+    // many channels
+    in_shapes[2] = {{
+      { 40, 40, 40, 3 },
+    }};
+
+    out_shapes[2] = {{
+      { 51, 51, 51, 3 },
+    }};
+  }
+
+  vector<ResamplingParams3D> GenerateParams(const TensorListShape<4> &out_shape,
+                                            const TensorListShape<4> &in_shape) {
+    vector<ResamplingParams3D> params;
+    params.resize(in_shape.num_samples());
+    std::bernoulli_distribution dist;
+    std::uniform_real_distribution<float> start_dist(0.05, 0.3);
+    std::uniform_real_distribution<float> end_dist(0.7, 0.95);
+    for (int i = 0; i < in_shape.num_samples(); i++) {
+      auto in_sample_shape = in_shape.tensor_shape_span(i);
+      for (int d = 0; d < 3; d++) {
+        params[i][d].min_filter = interp;
+        params[i][d].mag_filter = interp;
+        params[i][d].output_size = out_shape.tensor_shape_span(i)[d];
+        if (d == 2) {
+          do {
+            params[i][d].roi.use_roi = true;
+            params[i][d].roi.start = start_dist(rng) * in_sample_shape[d];
+            params[i][d].roi.end = end_dist(rng) * in_sample_shape[d];
+            if (dist(rng))
+              std::swap(params[i][d].roi.start, params[i][d].roi.end);
+          } while (interp == ResamplingFilterType::Nearest &&
+              !CheckNN(params[i][d].output_size, params[i][d].roi.start, params[i][d].roi.end));
+        }
+      }
+    }
+    return params;
+  }
+
+  // Checks for possible rounding problems leading to selecting different source pixel
+  // when running NN resampling.
+  static bool CheckNN(int size, float start, float end) {
+    float step = (end - start) / size;
+    float x = start + step * 0.5f;
+    for (int i = 0; i < size; i++, x += step) {
+      if (std::abs(x - std::floor(x)) < 0.01f)
+        return false;
+    }
+    return true;
+  }
+
+  void RunGPU() {
+    cudaStream_t stream = 0;
+
+    ResampleGPU<Out, In, 3> kernel;
+    KernelContext ctx;
+    ctx.gpu.stream = stream;
+    ScratchpadAllocator sa;
+    TestDataGenerator<In> tdg;
+    TestTensorList<In> in;
+    TestTensorList<Out> out, ref;
+
+    int niter = NumIter();
+    for (int iter = 0; iter < niter; iter++) {
+      const TensorListShape<4> &in_shape = in_shapes[iter];
+      int N = in_shape.num_samples();
+      in.reshape(in_shape);
+      for (int i = 0; i < N; i++)
+        tdg.GenerateTestData(in.gpu(stream)[i], 30, stream);
+
+      const TensorListShape<4> &out_shape = out_shapes[iter];
+
+      vector<ResamplingParams3D> params = GenerateParams(out_shape, in_shape);
+
+      Resample3Dvia2D(ref, in, make_span(params), stream);
+      auto ref_cpu = ref.cpu(stream);
+      auto ref_gpu = ref.template gpu<4>(stream);
+      // ref.invalidate_gpu();
```

Comment:
?

Reply:
I should re-enable it - I disabled this so I can dump the results to files.